src/Database/Driver.php

```diff
@@ -539,11 +546,61 @@ public function getMaxAliasLength(): ?int
     /**
      * @inheritDoc
      */
-    public function setLogger(?LoggerInterface $logger): void
+    public function getLogger(): ?LoggerInterface
+    {
+        return $this->logger;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setLogger(LoggerInterface $logger): void
     {
         $this->logger = $logger;
     }
 
+    /**
+     * Create logger instance.
+     *
+     * @param string|bool $log Log config.
+     * @return \Psr\Log\LoggerInterface
+     * @psalm-param string|true $log
+     */
+    protected function createLogger(string|bool $log): LoggerInterface
+    {
+        if ($log === true) {
+            $log = QueryLogger::class;
+        }
+
+        /** @var class-string<\Psr\Log\LoggerInterface>|null $className */
+        $className = App::className($log, 'Cake/Log', 'Log');
+        if ($className === null) {
+            throw new CakeException(
+                'For logging you must either set the `log` config to a FQCN which implemnts Psr\Log\LoggerInterface' .
+                ' or require the cakephp/log package in your composer config.'
+            );
+        }
+
+        return new $className();
+    }
+
+    /**
+     * Logs a query string using the configured logger object.
+     *
+     * @param string $sql String to be logged.
+     * @return void
+     */
+    protected function log(string $sql): void
```

Comment:
Can we call this `logQuery()` since it's explicitly meant to log a query string?

Reply:
I renamed it to `logSql()`

Reply:
Actually I changed the method a bit and named it `logQuery()` eventually :).